pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -590,6 +617,13 @@ var CinderVolumeFilter = VolumeFilter{
 		}
 		return "", false
 	},
+
+	IsMigrated: func(csiNode *storagev1beta1.CSINode) bool {
```

Comment:
I do not think this check implements KEP proposal. The KEP says:

>old predicates will be modified to perform an additional check of `CSINode` object at very beginning of their code. If they detect that `CSINode` object contains limit of same underlying volume type or CSI migration has been enabled for the volume, the old predicate will return early(with success) and `MaxCSIVolumeCountPred` will be responsible for counting both CSI and in-tree volumes of same type.


But what this check does is - it only defers to CSI predicate when migration is enabled *AND* `CSINode` object has volume limits for migrated driver. I think originally we wanted to ensure that, volume limits for both in-tree and CSI driver is counted correctly even if both are in-use on a node.

Reply:
I missed that, and it should be easy to fix but this also implies that the old predicates will defer to the CSI predicate if `CSINode` contains volume limits, even if the migration is disabled.

It means that the limits of an in-tree volume will change if someone installs a CSI driver (that happens to be the "migration counterpart" of the given in-tree volume).

Reply:
I still don't like moving limits between CSINode and Node back and forth as a driver is installed, I think that the KEP is wrong in this part. See @msau42's comment at https://github.com/kubernetes/enhancements/pull/942#discussion_r277060459, IMO her comment and the code in this PR it is the right thing to do.

Shall we fix the KEP?

Reply:
The reason for doing what we did was - to ensure that we count volumes correctly even if user is using CSI and in-tree volumes simultaneously in the cluster without explicitly enabling in-tree migration. Like imagine a user is using 20 in-tree EBS PVs and 30 CSI EBS PVs on a node, in which case if we kept volume counting to different predicates then we will not have accurate count. It was one of the goals.

Having said that - we can perhaps relax that requirement for now because I am not sure if mixed volumes use without explicitly enabling in-tree migration should be "supported" or not.

Reply:
The current (old) csi predicate already doesn't handle this case. I think we can consider supporting this use case if needed in the future, but I would prefer to keep the logic as simple as possible for now. It's already quite complex with all the cases that we need to handle.

This is only a temporary state while migration is not on yet.

Reply:
Yeah I am fine with dropping this requirement from KEP.

Reply:
Updated, PTAL.